node/cli/src/command.rs

```diff
@@ -470,6 +470,22 @@ pub fn run() -> Result<()> {
 				Err("Try-runtime must be enabled by `--features try-runtime`.".into())
 			}
 		},
+		Some(Subcommand::ExportRuntimeVersion(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			let chain_spec = &runner.config().chain_spec;
+			with_runtime_or_err!(chain_spec, {
+				{
+					runner.async_run(|config| {
+						// grab the task manager.
```

Comment:
doesn't the change spec already have the runtime version?

Reply:
Not sure what it means, with_runtime_or_err macro requires chain spec

Reply:
We cant do that as we are not invoking runtime_api but get it directly from file, to access native_runtime_api, i have to create client etc which is not needed for this case as we want to dump out runtime version

Chain spec is used by the macro to identity the current chain identity and import correct runtime code for the same, given we hide different runtimes behind different flags

Reply:
@enddynayn  i was refactoring some of it, let me know if all looks good now